src/R/Components/Impl/PackageManager/Implementation/RPackageManager.cs

```diff
@@ -27,6 +34,42 @@ internal class RPackageManager : IRPackageManager {
             return VisualComponent;
         }
 
+        public async Task<IList<RPackage>> GetInstalledPackagesAsync() {
+            if (_interactiveWorkflow.RSession == null || !_interactiveWorkflow.RSession.IsHostRunning) {
+                return new List<RPackage>();
+            }
+
+            REvaluationResult result;
+            using (IRSessionEvaluation eval = await _interactiveWorkflow.RSession.BeginEvaluationAsync()) {
```

Comment:
Ahh... It's a gotcha of the way the initialization code is structured currently. Basically when the session creates the host, it queues up the code that loads the package (via `BeginEvaluationAsync`, or rather some internal code that amounts to the same thing and uses the same queue) as a first eval request on the host. So any subsequent eval that goes through `BeginEvaluationAsync` will be next in queue, and initialization code is guaranteed to run by the time it happens. But when you call `EvaluateAsync` directly, it doesn't go in that queue - it just gets fired directly to the host.

@AlexanderSher, I think we'll need to discuss how to properly handle this. But for now let's keep `BeginEvaluationAsync`. It does work after all, it just means that package manager will not be accessible while REPL is running code. And switching to straight `EvaluateAsync` later, once we figure out how to solve this, is cheap and easy.